Content.Shared/Gibbing/Systems/GibbingSystem.cs

```diff
@@ -0,0 +1,309 @@
+using Robust.Shared.Audio.Systems;
+using Robust.Shared.Containers;
+using Robust.Shared.Map;
+using Robust.Shared.Physics.Systems;
+using Robust.Shared.Prototypes;
+using Robust.Shared.Random;
+
+namespace Content.Shared.Gibbing.Systems;
+
+public sealed class GibbingSystem : EntitySystem
+{
+    [Dependency] private readonly SharedContainerSystem _containerSystem = default!;
+    [Dependency] private readonly SharedTransformSystem _transformSystem = default!;
+    [Dependency] private readonly IPrototypeManager _prototypeManager = default!;
+    [Dependency] private readonly SharedAudioSystem _audioSystem = default!;
+    [Dependency] private readonly SharedPhysicsSystem _physicsSystem = default!;
+    [Dependency] private readonly IRobustRandom _random = default!;
+
+    //TODO: (future optimization) implement a system that "caps" giblet entities by deleting the oldest ones once we reach a certain limit, customizable via CVAR
+
+
+    /// <summary>
+    /// Attempt to gib a specified entity. That entity must have a gibable components. This method is NOT recursive will only
+    /// work on the target and any entities it contains (depending on gibContentsOption)
+    /// </summary>
+    /// <param name="outerEntity">The outermost entity we care about, used to place the dropped items</param>
+    /// <param name="target">Target entity we wish to gib</param>
+    /// <param name="gibType">What type of gibing are we performing</param>
+    /// <param name="gibContentsOption">What type of gibing do we perform on any container contents?</param>
+    /// <param name="droppedEntities">a hashset containing all the entities that have been dropped/created</param>
+    /// <param name="gibable">The gibable component</param>
+    /// <param name="randomSpreadMod">How much to multiply the random spread on dropped giblets(if we are dropping them!)</param>
+    /// <param name="playAudio">Should we play audio</param>
+    /// <param name="allowedContainers">A list of containerIds on the target that permit gibing</param>
+    /// <param name="excludedContainers">A list of containerIds on the target that DO NOT permit gibing</param>
+    /// <param name="launchCone">The cone we are launching giblets in (if we are launching them!)</param>
+    /// <param name="launchGibs">Should we launch giblets or just drop them</param>
+    /// <param name="launchDirection">The direction to launch giblets (if we are launching them!)</param>
+    /// <param name="launchImpulse">The impluse to launch giblets at(if we are launching them!)</param>
+    /// /// <param name="logMissingGibable">Should we log if we are missing a gibbableComp when we call this function</param>
+    /// <param name="launchImpulseVariance">The variation in giblet launch impulse (if we are launching them!)</param>
+    /// <returns>True if successful, false if not</returns>
+    public bool TryGibEntity(EntityUid outerEntity, EntityUid target, GibType gibType, GibContentsOption gibContentsOption,
```

Comment:
Huh, I wasn't aware you could do that. I thought that Entity<T> didn't support generics. I just double checked with Smug and yeah that would work. I'm going to refactor all the signatures to use that then :)

Reply:
In regards to making the signature cleaner, that isn't really an option due to the complexity. Most of the arguments are optional and have default values anyways so you can use (Parameter:value) for arguments instead of filling out the whole arguments.